vllm/utils/flashinfer.py

```diff
@@ -106,6 +110,73 @@ def has_flashinfer_cutlass_fused_moe() -> bool:
     return True
 
 
+@functools.cache
+def has_nvidia_artifactory() -> bool:
+    """Return ``True`` if NVIDIA's artifactory is accessible.
+    
+    This checks connectivity to the kernel inference library artifactory
+    which is required for downloading certain cubin kernels like TRTLLM FHMA.
+    """
+    # Referenced from https://github.com/flashinfer-ai/flashinfer/blob/0c9a92c3d9a7e043ab6f3f7b2273269caf6ab044/flashinfer/jit/cubin_loader.py#L35  # noqa: E501
+    artifactory_url = "https://edge.urm.nvidia.com/artifactory/sw-kernelinferencelibrary-public-generic-local/"  # noqa: E501
```

Comment:
Using this absolute online address verification may affect some offline vllm instances? There is no solution to modify this address through environment variables like flashinfer does.